R/loopy.R

```diff
@@ -70,36 +76,105 @@ loopy <- function(resamples, grid, static) {
         current_wflow <- remove_log_notes(current_wflow)
       }
 
-      num_pred_iter <- nrow(current_model$predict_stage[[1]])
       current_grid <- rebind_grid(current_pre, current_model)
 
-      # ------------------------------------------------------------------------
-      # Iterate over predictions and postprocessors
+      has_submodel <- has_sub_param(current_model$predict_stage[[1]])
+      num_prd_iter <- nrow(current_model$predict_stage[[1]])
 
-      pred <- .catch_and_log(
-        predictions(
-          wflow_current = current_wflow,
-          sched = current_model,
-          static = static,
-          grid = current_grid
-        )
-      )
-      if (has_log_notes(pred)) {
-        location <- glue::glue("prediction {mod}/{num_mod_iter}")
-        notes <- append_log_notes(notes, pred, location)
-        if (is_failure(pred)) {
-          next
+      # --------------------------------------------------------------------------
+      # Iterate over prediction submodels
+
+      for (prd in seq_len(num_prd_iter)) {
+
+        # cli::cli_inform("Predicting {prd} of {num_prd_iter}")
+
+        current_prd <- current_model$predict_stage[[1]][prd, ]
+
+        if (has_submodel) {
+          sub_nm <- get_sub_param(current_prd)
+          sub_grid <- current_prd[, sub_nm]
+
+          # The assigned submodel parameter (from min_grid()) is in the
+          # current grid. Remove that and add the one that we are predicting on
+
+          current_grid <- current_grid |>
```

Comment:
Predicts one submodel at a time. We _could_ use `multi_predict()` to get those all at once and loop over that but this is more straightforward. We can benchmark if we are worried about there being a difference.